testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusSiteTest.java

```diff
@@ -32,67 +33,71 @@
     public static final String webPageUrl = Commands.getCodeQuarkusURL("https://code.quarkus.redhat.com/");
     public static final String elementTitleByText = "Quarkus - Start coding with code.quarkus.redhat.com";
     public static final String elementIconByXpath = "//link[@rel=\"shortcut icon\"][@href=\"https://www.redhat.com/misc/favicon.ico\"]";
     public static final String elementRedHatLogoByXpath= "//img[@class=\"logo\"][@alt=\"Red Hat Logo\"]";
     public static final String elementSupportedFlagByXpath = "//div[@class=\"extension-tag supported dropdown-toggle\"]";
-    public static final String elementQuarkusPlatformVersionByXpath = "normalize-space(//div[@class=\"quarkus-stream final\"]/@title)";
+    public static final String elementQuarkusPlatformVersionByXpath = "//div[@class=\"quarkus-stream final\"]";
 
-    private WebClient webClient;
+    private BrowserContext browserContext; // Operates in incognito mode
 
-    @BeforeEach
+    @BeforeAll
     public void init(){
-        webClient = new WebClient(BrowserVersion.CHROME);
-        webClient.getOptions().setThrowExceptionOnScriptError(false);
-        webClient.getOptions().setJavaScriptEnabled(true);
-        java.util.logging.Logger.getLogger("com.gargoylesoftware").setLevel(Level.OFF);
+        Playwright playwright = Playwright.create();
+        Browser browser = playwright.chromium().launch(new BrowserType.LaunchOptions()
+                .setArgs(List.of("--headless", "--disable-gpu", "--no-sandbox")));
+        browserContext = browser.newContext();
+        LOGGER.info("Incognito browser session has been created");
     }
 
-    @AfterEach
+    @AfterAll
     public void close(){
-        webClient.close();
+        browserContext.close();
     }
 
 
     @Test
-    public void validatePresenceOfIcon(TestInfo testInfo) throws Exception {
-        final HtmlPage page = loadPage(webPageUrl, 60);
+    public void validatePresenceOfIcon(TestInfo testInfo) {
+        Page page = loadPage(webPageUrl, 60);
         LOGGER.info("Trying to find element: " + elementIconByXpath);
-        HtmlElement shortcutIcon = page.getFirstByXPath(elementIconByXpath);
-        assertNotNull(shortcutIcon, "Element: " + elementIconByXpath + " is missing!");
+        Locator icon = page.locator(elementIconByXpath);
+        assertEquals(1, icon.count(), "Element: " + elementIconByXpath + " is missing!");
     }
 
     @Test
-    public void validatePresenceOfTitle(TestInfo testInfo) throws Exception {
-        final HtmlPage page = loadPage(webPageUrl, 60);
+    public void validatePresenceOfTitle(TestInfo testInfo) {
+        Page page = loadPage(webPageUrl, 60);
         LOGGER.info("Verify page title is: " + elementTitleByText);
-        assertEquals(page.getTitleText(), elementTitleByText,
-                "Title doesn't match. Found on the web: " + page.getTitleText() + ". Expected: " + elementTitleByText);
+        assertEquals(page.title(), elementTitleByText,
```

Comment:
```suggestion
        assertEquals(elementTitleByText, page.title(),
```
`Expected` and `actual` are swapped.